src/plugins/discover/public/application/main/hooks/use_esql_mode.ts

```diff
@@ -74,6 +74,36 @@ export function useEsqlMode({
             return;
           }
 
+          // We need to reset the default profile state on index pattern changes
+          // when loading starts to ensure the correct pre fetch state is available
+          // before data fetching is triggered
+          if (next.fetchStatus === FetchStatus.LOADING) {
```

Comment:
This whole file should really be integrated into the main data fetching routine to avoid hackiness like this... Another refactoring opportunity.